### ISSUE TITLE
`*_SetsALPN` tests fail on Windows.81.Amd64.Open

### ISSUE THREAD

**Issue by dougbu** [dotnet/aspnetcore#34562]
## Failing Test(s)

- `Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.ServerOptionsSelectionCallback_SetsALPN()`
- `Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.TlsHandshakeCallbackOptionsOverload_SetsALPN()`

## Error Message

```text
Assert.Equal() Failure
Expected: h2
Actual:   http/1.1
```

and

``` text
Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware Information: HTTP/2 over TLS is not supported on Windows versions older than Windows 10 and Windows Server 2016 due to incompatible ciphers or missing ALPN support. Falling back to HTTP/1.1 instead.
```

## Stacktrace

<details>
<summary>ServerOptionsSelectionCallback_SetsALPN()</summary>

```text
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.ServerOptionsSelectionCallback_SetsALPN() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:line 820
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.ServerOptionsSelectionCallback_SetsALPN() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:line 820
--- End of stack trace from previous location ---
```
</details>

<details>
<summary>TlsHandshakeCallbackOptionsOverload_SetsALPN()</summary>

```text
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.TlsHandshakeCallbackOptionsOverload_SetsALPN() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:line 852
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.TlsHandshakeCallbackOptionsOverload_SetsALPN() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:line 852
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
 | [0.000s] TestLifetime Information: Starting test TlsHandshakeCallbackOptionsOverload_SetsALPN at 2021-07-19T13:00:40
 | [0.002s] Microsoft.Extensions.Hosting.Internal.Host Debug: Hosting starting
 | [0.002s] Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware Information: HTTP/2 over TLS is not supported on Windows versions older than Windows 10 and Windows Server 2016 due to incompatible ciphers or missing ALPN support. Falling back to HTTP/1.1 instead.
 | [0.003s] Microsoft.Hosting.Lifetime Information: Now listening on: https://127.0.0.1:0
 | [0.003s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Loaded hosting startup assembly InMemory.FunctionalTests
 | [0.003s] Microsoft.Hosting.Lifetime Information: Application started. Press Ctrl+C to shut down.
 | [0.003s] Microsoft.Hosting.Lifetime Information: Hosting environment: Production
 | [0.003s] Microsoft.Hosting.Lifetime Information: Content root path: C:\h\w\B30209FF\w\B8ED09D5\e\
 | [0.003s] Microsoft.Extensions.Hosting.Internal.Host Debug: Hosting started
 | [0.015s] Microsoft.AspNetCore.Server.Kestrel.Connections Debug: Connection id "0HMAAMKPPGC2H" accepted.
 | [0.016s] Microsoft.AspNetCore.Server.Kestrel.Connections Debug: Connection id "0HMAAMKPPGC2H" started.
 | [0.026s] Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware Debug: Connection "0HMAAMKPPGC2H" established using the following protocol: Tls12
 | [0.034s] Microsoft.Extensions.Hosting.Internal.Host Debug: Hosting stopping
 | [0.034s] Microsoft.AspNetCore.Server.Kestrel.Connections Debug: Connection id "0HMAAMKPPGC2H" disconnecting.
 | [0.034s] Microsoft.Hosting.Lifetime Information: Application is shutting down...
 | [0.034s] Microsoft.AspNetCore.Server.Kestrel.Connections Debug: Connection id "0HMAAMKPPGC2H" stopped.
 | [0.034s] Microsoft.Extensions.Hosting.Internal.Host Debug: Hosting stopped
 | [0.096s] Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests Error: Test threw an exception.
 | Xunit.Sdk.EqualException: Assert.Equal() Failure
 | Expected: h2
 | Actual:   http/1.1
 |    at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 40
 |    at Xunit.Assert.Equal[T](T expected, T actual) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 24
 |    at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.TlsHandshakeCallbackOptionsOverload_SetsALPN() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:line 852
 |    at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.HttpsConnectionMiddlewareTests.TlsHandshakeCallbackOptionsOverload_SetsALPN() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:line 852
 |    at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 264
 | --- End of stack trace from previous location ---
 |    at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
 |    at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
 | [0.100s] TestLifetime Information: Finished test TlsHandshakeCallbackOptionsOverload_SetsALPN in 0.0993653s
```
</details>

## Build

Failing consistently in aspnetcore-helix-matrix runs e.g.
- https://dev.azure.com/dnceng/public/_build/results?buildId=1245614&view=results
- https://dev.azure.com/dnceng/public/_build/results?buildId=1245103&view=results

**Comment by dougbu** [comment]
Fixed w/ 4c0cb774815a

**Comment by dougbu** [comment]
@BrennanConroy @Tratcher was this the only issue tracking the problem on Win8.1❔ I'm trying to decide between "dupe" and "done"😺
